core/common/src/main/java/alluxio/util/URIUtils.java

```diff
@@ -50,6 +50,22 @@ public static URI appendPath(URI base, String path) throws URISyntaxException {
   }
 
   /**
+   * Appends the given path to the given base URI. It throws an {@link RuntimeException} if
+   * the inputs are malformed.
+   *
+   * @param base the base URI
+   * @param path the path to append
+   * @return the URI resulting from appending the base and the path
+   */
+  public static URI appendPathOrDie(URI base, String path) {
```

Comment:
Let's keep the current version in this PR and consolidate `appendPathOrDie` with `appendPath` in a separate PR